examples/jsm/webxr/XREstimatedLight.js

```diff
@@ -0,0 +1,225 @@
+import {
+	DirectionalLight,
+	Group,
+	LightProbe,
+	WebGLCubeRenderTarget
+} from "../../../build/three.module.js";
+
+class SessionLightProbe {
+
+	constructor( xrLight, renderer, lightProbe, estimationStartCallback ) {
+
+		this.xrLight = xrLight;
+		this.renderer = renderer;
+		this.lightProbe = lightProbe;
+		this.xrWebGLBinding = null;
+		this.estimationStartCallback = estimationStartCallback;
+		this.frameCallback = this.onXRFrame.bind( this );
+
+		const session = renderer.xr.getSession();
+
+		// If the XRWebGLBinding class is available then we can also query an
+		// estimated reflection cube map.
+		if ( 'XRWebGLBinding' in window ) {
+
+			const gl = renderer.getContext();
+
+			// Ensure that we have any extensions needed to use the preferred cube map format.
+			switch ( session.preferredReflectionFormat ) {
+
+				case 'srgba8':
+					gl.getExtension( 'EXT_sRGB' );
+					break;
+
+				case 'rgba16f':
+					gl.getExtension( 'OES_texture_half_float' );
+					break;
+
+			}
+
+			this.xrWebGLBinding = new XRWebGLBinding( session, gl );
+
+			// TODO: This should actually use the 'reflectionchange' event instead
+			// but there is a known issue in Chrome with that at the moment.
+			/*this.lightProbe.addEventListener('reflectionchange', () => {
+				this.updateReflection();
+			});*/
+
+			// Until the above is fixed, update the reflection once a second. (Which is
+			// approximately how frequently the event would fire anyway).
+			setInterval( () => {
+
+				this.updateReflection();
```

Comment:
Does updating the environment have a cost in the event we aren't using it? And is there cost on the JS side, or only on the implementation side?

Reply:
Good point! There is a cost to updating it, which shouldn't be frame-droppingly severe but if your app wasn't using it (like you suggest below) then it's also wasted effort. We could turn the environment map into a getter attrib and only start updating it if you query it as a start, but that doesn't provide an easy way to stop using it, and it could accidentally get triggered by things like iterating over the object properties.

Reply:
Might be worth just setting an input parameter to use it or not.